core/trino-main/src/main/java/io/trino/type/LikePatternType.java

```diff
@@ -52,13 +55,27 @@ public Object getObject(Block block, int position)
             return null;
         }
 
-        return joniRegexp(block.getSlice(position, 0, block.getSliceLength(position)));
+        String pattern = block.getSlice(position, 0, block.getSliceLength(position)).toStringUtf8();
+        Optional<Character> escape = Optional.empty();
+        if (!block.isNull(position + 1)) {
+            escape = Optional.of((char) block.getInt(position + 1, 0));
+        }
+
+        return LikeMatcher.compile(pattern, escape);
```

Comment:
It should work. The pattern gets written to a block as `string | int?` for the pattern and the optional escape. But maybe you're seeing something I'm not?

In any case, LikePatternType is a synthetic type and this method is only used to serialize / deserialize patterns to ship them to workers.

Reply:
`getObject(position)` doesn’t us you the LikePattern  at position without caller side manipulation. Also, we add 2 or 3 elements respectively in case of  with/without escape. (appendNull.closeEntry adds two vs writeInt.closeEntry adds one) so position manipulation may not be straightforward. I see the point that this is all synthetic/internal, but curious if the usage for more than one elements is kosher

Reply:
```
        BlockBuilder blockBuilder = LIKE_PATTERN.createBlockBuilder(new PageBuilderStatus().createBlockBuilderStatus(), 10);
        LIKE_PATTERN.writeObject(blockBuilder, LikeMatcher.compile("abcd", Optional.of('g')));
        LIKE_PATTERN.writeObject(blockBuilder, LikeMatcher.compile("abc"));
        Block block = blockBuilder.build(); -- contains 5 positions (2 + 3) 
        //  incorrect mapping to previous entry's escape. 
        LIKE_PATTERN.getObject(block, 1);
```

Reply:
Oh, yeah, that's completely broken. I actually meant it to write at different offsets but somehow I ended using positions instead. I'll fix it.